### ISSUE TITLE
Audio recording not working; could you document the requirements of gstreamer pipelines?

### ISSUE THREAD

**Issue by crquan** [iacopodeenosee/EasyScreenCast#47]
on a branch new ubuntu-gnome 15.10 installation, the audio recording is not working, I suspect is is the gstreamer pipeline, here is the default gstreamer plugins; could you document the requirements? and how to debug the audio recording problems? would be even better if you have one line of `apt-get install ` required packages

```gstreamer
queue ! videorate
 ! vp9enc min_quantizer=13 max_quantizer=13 cpu-used=5 deadline=1000000 threads=%T
 ! queue ! mux. pulsesrc ! queue ! audioconvert ! vorbisenc ! queue
 ! mux. webmmux name=mux 
```

**Comment by iacopodeenosee** [comment]
> how to debug the audio recording problems?

you must **enable verbose log** into option window, and then view it by the log application or by terminal with execute this command: ``` journalctl --since=today --no-pager | grep JS ```

> could you document the requirements?

it use with introspection Gio lib, GLib lib, Gst lib, Gvc lib; all these libraries are already included in the installation of a gnome, in particular [gstreamer](http://packages.ubuntu.com/wily/gstreamer1.0-libav) is already installed in ubuntu 15.10 as a dependency of a [gnome metapackage](http://packages.ubuntu.com/wily/gnome).
if you need help paste the log here, thx.

**Comment by crquan** [comment]
the ubuntu-gnome 15.10 has gstreamer installed and no problem to run google hangout video/audio chat, so the system itself has no problem on audio recording,

but the problem here with EasyScreenCast is the generated webm file has only 1 sec of audio, all remaining is silent; the screen cast recorded has no problem, just only 1 sec of sound

```log
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > stream gnome recorder captured
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream index: 6
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream card index: -1
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _application_ID: null
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream name: GNOME Shell
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream icon: audio-input-microphone
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream description: Record Stream
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream port: null
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream add - ID: 2238
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > stream gnome recorder captured
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream index: 6
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream card index: -1
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _application_ID: null
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream name: GNOME Shell
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream icon: audio-input-microphone
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream description: Record Stream
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream port: null
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream add - ID: 2238
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > stream gnome recorder captured
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream index: 6
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream card index: -1
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _application_ID: null
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream name: GNOME Shell
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream icon: audio-input-microphone
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream description: Record Stream
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream port: null
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream add - ID: 2238
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > stream gnome recorder captured
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream index: 6
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream card index: -1
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _application_ID: null
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream name: GNOME Shell
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream icon: audio-input-microphone
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream description: Record Stream
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream port: null
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream add - ID: 2238
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > stream gnome recorder captured
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream index: 6
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream card index: -1
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _application_ID: null
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream name: GNOME Shell
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream icon: audio-input-microphone
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream description: Record Stream
Nov 08 16:12:23 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > _stream port: null
Nov 08 16:12:24 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 1
Nov 08 16:12:25 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 2
Nov 08 16:12:26 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > refresh indicator -A true -F true
Nov 08 16:12:26 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 3
Nov 08 16:12:27 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > refresh indicator -A true -F false
Nov 08 16:12:27 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 4
Nov 08 16:12:28 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 5
Nov 08 16:12:29 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 6
Nov 08 16:12:30 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 7
Nov 08 16:12:31 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 8
Nov 08 16:12:32 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 9
Nov 08 16:12:33 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 10
Nov 08 16:12:34 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 11
Nov 08 16:12:35 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 12

Nov 08 16:12:36 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 13
Nov 08 16:12:37 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 14
Nov 08 16:12:38 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 15
Nov 08 16:12:39 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 16
Nov 08 16:12:40 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > continued TimerCounting | sec: 17
Nov 08 16:12:41 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > intercept key combination
Nov 08 16:12:41 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > stop recording
Nov 08 16:12:41 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > stop video recording
Nov 08 16:12:41 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > refresh indicator -A false -F undefined
Nov 08 16:12:41 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > screencast stop - true
Nov 08 16:12:41 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > finish TimerCounting
Nov 08 16:12:41 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > stop TimerCounting called
Nov 08 16:14:57 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream remove - ID: 2238
Nov 08 16:14:57 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream remove - ID: 2238
Nov 08 16:14:57 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream remove - ID: 2238
Nov 08 16:14:57 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream remove - ID: 2238
Nov 08 16:14:57 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream remove - ID: 2238
Nov 08 16:14:57 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream remove - ID: 2238
Nov 08 16:14:57 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream remove - ID: 2238
Nov 08 16:14:57 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream remove - ID: 2238
Nov 08 16:14:57 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream remove - ID: 2238
Nov 08 16:14:57 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream remove - ID: 2238
Nov 08 16:14:57 gnome3-on-mac gnome-session[22518]: Gjs-Message: JS LOG: ESC > mixer stream remove - ID: 2238

```

**Comment by crquan** [comment]
and sometimes it comes with high cpu usage of gnome-shell, I'm not sure if related

```log
mint@gnome3-on-mac:~$ ps -L -o pid,lwp,nlwp,pcpu,pmem,vsz,rss,comm,wchan:26,cmd 9148
  PID   LWP NLWP %CPU %MEM    VSZ   RSS COMMAND         WCHAN                      CMD
 9148  9148    7  1.5  4.4 2701152 735228 gnome-shell   poll_schedule_timeout      /usr/bin/gnome-shell
 9148  9151    7  0.0  4.4 2701152 735228 gmain         poll_schedule_timeout      /usr/bin/gnome-shell
 9148  9160    7  0.0  4.4 2701152 735228 gdbus         poll_schedule_timeout      /usr/bin/gnome-shell
 9148  9162    7  0.0  4.4 2701152 735228 dconf worker  poll_schedule_timeout      /usr/bin/gnome-shell
 9148  9165    7  0.0  4.4 2701152 735228 threaded-ml   poll_schedule_timeout      /usr/bin/gnome-shell
 9148  9166    7  0.0  4.4 2701152 735228 JS GC Helper  futex_wait_queue_me        /usr/bin/gnome-shell
 9148  9167    7  0.0  4.4 2701152 735228 JS Sour~ Thread futex_wait_queue_me      /usr/bin/gnome-shell
```

**Comment by iacopodeenosee** [comment]
i think that the VP9 encoding is too heavy, try this custom pipeline:
```queue ! videorate ! vp8enc min_quantizer=13 max_quantizer=13 cpu-used=5 deadline=1000000 threads=%T ! queue ! mux. pulsesrc ! queue ! audioconvert ! vorbisenc ! queue ! mux. webmmux name=mux ```
in this it use a VP8 encoding, if this works well i change the default pipeline to vp8. thx

**Comment by eychenz** [comment]
On fedora 23 Gnome 3.18
I experienced the same problem.
Using vp8enc solved the problem. thnx

**Comment by crquan** [comment]
this reproduced gnome-shell's high cpu usage after screencast for one minute, and after a long time (5 minutes) the cpu usage doesn't drop

```console
mint@gnome3-on-mac:~ 05:08:43 ➸ ps -L -o pid,lwp,nlwp,pcpu,pmem,vsz,rss,comm,wchan:26,cmd 6207
  PID   LWP NLWP %CPU %MEM    VSZ   RSS COMMAND         WCHAN                      CMD
 6207  6207   21  1.7 35.3 7830524 5773236 gnome-shell  poll_schedule_timeout      /usr/bin/gnome-shell
 6207  6214   21  0.0 35.3 7830524 5773236 gmain        poll_schedule_timeout      /usr/bin/gnome-shell
 6207  6221   21  0.0 35.3 7830524 5773236 gdbus        poll_schedule_timeout      /usr/bin/gnome-shell
 6207  6223   21  0.0 35.3 7830524 5773236 dconf worker poll_schedule_timeout      /usr/bin/gnome-shell
 6207  6225   21  0.0 35.3 7830524 5773236 threaded-ml  poll_schedule_timeout      /usr/bin/gnome-shell
 6207  6227   21  0.0 35.3 7830524 5773236 JS GC Helper futex_wait_queue_me        /usr/bin/gnome-shell
 6207  6228   21  0.0 35.3 7830524 5773236 JS Sour~ Thread futex_wait_queue_me     /usr/bin/gnome-shell
 6207 17880   21  1.7 35.3 7830524 5773236 threaded-ml  poll_schedule_timeout      /usr/bin/gnome-shell
 6207 17881   21  0.0 35.3 7830524 5773236 queue1:src   futex_wait_queue_me        /usr/bin/gnome-shell
 6207 17882   21  0.0 35.3 7830524 5773236 queue3:src   futex_wait_queue_me        /usr/bin/gnome-shell
 6207 17883   21 57.8 35.3 7830524 5773236 queue0:src   ?                          /usr/bin/gnome-shell
 6207 17884   21  0.0 35.3 7830524 5773236 queue2:src   futex_wait_queue_me        /usr/bin/gnome-shell
 6207 17885   21  0.0 35.3 7830524 5773236 pulsesrc0:src futex_wait_queue_me       /usr/bin/gnome-shell
 6207 17886   21  4.0 35.3 7830524 5773236 shellrecordersr futex_wait_queue_me     /usr/bin/gnome-shell
 6207 17889   21  0.9 35.3 7830524 5773236 audiosrc-ringbu futex_wait_queue_me     /usr/bin/gnome-shell
 6207 17890   21 84.0 35.3 7830524 5773236 queue0:src   ?                          /usr/bin/gnome-shell
 6207 17891   21 62.4 35.3 7830524 5773236 queue0:src   ?                          /usr/bin/gnome-shell
 6207 17892   21 52.1 35.3 7830524 5773236 queue0:src   ?                          /usr/bin/gnome-shell
 6207 17893   21 60.4 35.3 7830524 5773236 queue0:src   ?                          /usr/bin/gnome-shell
 6207 17894   21 57.7 35.3 7830524 5773236 queue0:src   ?                          /usr/bin/gnome-shell
 6207 17895   21 43.1 35.3 7830524 5773236 queue0:src   ?                          /usr/bin/gnome-shell
```

**Comment by crquan** [comment]
Update: in longer time the high cpu usage dropped, I didn't record for how long time. after dropped it runs normally 1.9% of CPU usage and 353MB of physical memory.

```console
mint@gnome3-on-mac:~ 05:22:40 ➸ ps -L -o pid,lwp,nlwp,pcpu,pmem,vsz,rss,sgi_p,psr,comm,wchan:26,cmd 6207
  PID   LWP NLWP %CPU %MEM    VSZ   RSS P PSR COMMAND         WCHAN                      CMD
 6207  6207    7  1.8  2.1 2780044 351848 * 3 gnome-shell     poll_schedule_timeout      /usr/bin/gnome-shell
 6207  6214    7  0.0  2.1 2780044 351848 * 0 gmain           poll_schedule_timeout      /usr/bin/gnome-shell
 6207  6221    7  0.0  2.1 2780044 351848 * 6 gdbus           poll_schedule_timeout      /usr/bin/gnome-shell
 6207  6223    7  0.0  2.1 2780044 351848 * 6 dconf worker    poll_schedule_timeout      /usr/bin/gnome-shell
 6207  6225    7  0.0  2.1 2780044 351848 * 4 threaded-ml     poll_schedule_timeout      /usr/bin/gnome-shell
 6207  6227    7  0.0  2.1 2780044 351848 * 5 JS GC Helper    futex_wait_queue_me        /usr/bin/gnome-shell
 6207  6228    7  0.0  2.1 2780044 351848 * 1 JS Sour~ Thread futex_wait_queue_me        /usr/bin/gnome-shell
```

While, during the recording the gnome-shell's cpu usage is like 75% percentage of the all cpu cores in overall, actually above ps command tells 6 threads (from 21) of gnome-shell are running all the time, Virtual memory (address space) is 7.8GB, and Rss tells physical memory in use is 5.8GB.

Are these numbers too high?

**Comment by crquan** [comment]
after changed to vp8, it improves somewhat; but sound still sometimes stuck, intermittent, not continuous playing

**Comment by crquan** [comment]
on ubuntu the `/usr/bin/gst-launch` command is provided by gstreamer0.10-tools, and there is another `/usr/bin/gst-launch-1.0` provided by gstreamer-tools-1.0 that is confusing; I then removed `gstreamer0.10-tools gstreamer0.10-pulseaudio libgstreamer0.10-0` now it doesn't have packages dependency problem.

the `/usr/lib/x86_64-linux-gnu/gstreamer-1.0/libgstvpx.so` is provided by `gst-plugins-good1.0` package which is default installed.

**Comment by iacopodeenosee** [comment]
> after changed to vp8, it improves somewhat; but sound still sometimes stuck, intermittent, not continuous playing

strange, try to execute this command into a terminal, to end recording press **CTRL + C**

```gst-launch-1.0 --gst-debug=3  ximagesrc ! videoconvert ! queue ! videorate ! vp8enc  min_quantizer=10 max_quantizer=10 cpu-used=5 deadline=1000000  threads=2 ! queue ! mux. pulsesrc ! queue !  audioconvert ! vorbisenc ! queue ! mux. webmmux name=mux ! filesink location=screencast.webm```

in that log there is maybe something useful to define the problem; however i change the default pipeline to VP8 when is enabled the sound recording and close this issue because I think the problem is in the gnome shell recorder and not in the extension.
feel free to open other issue. thx

**Comment by crquan** [comment]
> gst-launch-1.0 --gst-debug=3 ximagesrc ! videoconvert ! queue ! videorate ! vp8enc min_quantizer=10 max_quantizer=10 cpu-used=5 deadline=1000000 threads=2 ! queue ! mux. pulsesrc ! queue ! audioconvert ! vorbisenc ! queue ! mux. webmmux name=mux ! filesink location=screencast.webm

this command is much better, I recorded it for one minute, no sound loss, and cpu / mem usage by gnome-shell remained at 2% and 400MB Rss, no much difference with idling.

```log
mint@gnome3-on-mac:~/Videos$ gst-launch-1.0 --gst-debug=3 ximagesrc ! videoconvert ! queue ! videorate ! vp8enc min_quantizer=10 max_quantizer=10 cpu-used=5 deadline=1000000 threads=2 ! queue ! mux. pulsesrc ! queue ! audioconvert ! vorbisenc ! queue ! mux. webmmux name=mux ! filesink location=screencast.webm
Setting pipeline to PAUSED ...
Pipeline is live and does not need PREROLL ...
0:00:00.014571465  2248       0xcedf70 FIXME                default gstutils.c:3766:gst_pad_create_stream_id_internal:<pulsesrc0:src> Creating random stream-id, consider implementing a deterministic way of creating a stream-id
0:00:00.014601365  2248       0xd25370 FIXME                default gstutils.c:3766:gst_pad_create_stream_id_internal:<ximagesrc0:src> Creating random stream-id, consider implementing a deterministic way of creating a stream-id
Setting pipeline to PLAYING ...
New clock: GstPulseSrcClock
Redistribute latency...
Redistribute latency...
0:00:00.039728245  2248       0xd25370 WARN               videopool gstvideopool.c:216:video_buffer_pool_set_config:<videobufferpool1> no caps in config
0:00:00.059157501  2248       0xd25370 ERROR             GST_BUFFER gstbuffer.c:558:gst_buffer_copy_into: failed to copy meta 0x7f5094003588 of API type GstMetaXImageSrcAPI
0:00:00.067004253  2248       0xd25230 WARN            audioencoder gstaudioencoder.c:970:gst_audio_encoder_finish_frame:<vorbisenc0> Can't copy metadata because input buffer disappeared
0:00:00.259615714  2248       0xd25280 FIXME               basesink gstbasesink.c:3117:gst_base_sink_default_event:<filesink0> stream-start event without group-id. Consider implementing group-id handling in the upstream elements
^Chandling interrupt.
Interrupt: Stopping pipeline ...
Execution ended after 0:00:18.613787295
Setting pipeline to PAUSED ...
0:00:18.630365433  2248 0x7f50a00034f0 WARN                audiosrc gstaudiosrc.c:244:audioringbuffer_thread_func:<pulsesrc0> error reading data -1 (reason: Resource temporarily unavailable), skipping segment
Setting pipeline to READY ...
Setting pipeline to NULL ...
Freeing pipeline ...
```

**Comment by crquan** [comment]
> gst-launch-1.0 --gst-debug=3 ximagesrc ! videoconvert ! queue ! videorate ! vp8enc min_quantizer=10 max_quantizer=10 cpu-used=5 deadline=1000000 threads=2 ! queue ! mux. pulsesrc ! queue ! audioconvert ! vorbisenc ! queue ! mux. webmmux name=mux ! filesink location=screencast.webm

I can use this command line as screen recording now; but the webm file recorded in this way seems have a length problem: the length is 1 second shown in totem player, the progress bar has proceeded to end, while it is still playing, it's able to play till the end; maybe it needs some post-processing to fix length? what's a proper way to tell gst-launch to finish recording? if not Ctrl-C to interrupt.

Just report another issue with EasyScreenCast:
1. In Option window the `File screencast` seems always showing my HOME directory, although I have changed it, and it saves file into the correct location, but this option window seems not showing the location I chose, it's always showing the default HOME
2. File size concerns: for 1.5 minutes recording, I get a 25MB file, it's because recording at my screen resolution (2880x1800), I am not sure what will be the file size if I need to record 1hr long? is there anyway to reduce file size by reduce resolution? My use case is to post the screencast to youtube afterwards, most probably 1080p or 720p in width would be sufficient.

**Comment by crquan** [comment]
@iacopodeenosee why is this closed? as I reported, use `gst-launch-1.0 ...` works and use this EasyScreenCast has broken audio, and high cpu / mem usage problem.

**Comment by iacopodeenosee** [comment]
> why is this closed?

because I not think it makes sense to use an external command :) , this extension use the integrate gnome shell recorder, and It was created to simplify the use of this and [waiting for the gnome team build the application designated for this] (https://wiki.gnome.org/Design/Apps/Potential/Recorder)... 

>In Option window the File screencast seems always showing my HOME directory, although I have changed it, and it saves file into the correct location, but this option window seems not showing the location I chose, it's always showing the default HOME

if I remember well the file selector widget is buggy and does not allow me to select a folder at the start, when I'll just make the options window will change the widget as suggested here #27 
at this moment to avoid these misunderstandings there is a label with the path of the selected folder above the widget

> File size concerns: for 1.5 minutes recording, I get a 25MB file, it's because recording at my screen resolution (2880x1800), I am not sure what will be the file size if I need to record 1hr long? is there anyway to reduce file size by reduce resolution? My use case is to post the screencast to youtube afterwards, most probably 1080p or 720p in width would be sufficient.

the best way to reduce/improve the quality of the recording is change the gstreamer pipeline, at this moment is a working in progress #30
